### PR TITLE
Adding documentation for the development stack

### PR DESCRIPTION
**Pull request by FunkySayu**

### CODE REVIEW

**Review by cassand — APPROVED** [review]

File: configure [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    else
+        Log::Notice "Skipping Tensorflow configuration. If you want to run it"
+        Log::Notice "again, run: rm third_party/tensorflow/.configured"
+    fi
+    Log::Success "Done configuring tensorflow"
+
+    # Install python dependencies
+    # TODO(funkysayu): Take a look if we can remove those.
+    Log::Notice "Installing third-party dependencies"
+    pip install --user numpy tensorflow grpcio
```
Comment (cassand) [inline configure]: tensorflow is no longer needed

**Review by cassand — CHANGES_REQUESTED** [review]

File: configure [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+source "$(dirname "${BASH_SOURCE[0]}")/devutils/lib/bash/logging.bash"
+
+
+main () {
+    cd "$(dirname "${BASH_SOURCE[0]}")"
+    [ ! -f "WORKSPACE" ] && Log::Fatal "unable to find the workspace directory."
+
+    # Fetch and configure dependencies
+    git submodule update
```
Comment (ArchangelX360) [inline configure]: --init needed on first update

File: docs/dev-stack.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ - generation of the `docker-compose.yml` file.
+
+# Pre-build requirements
+
+## System requirements
+
+To build the stack, we expect you to have the following tools on your system:
+ - [Docker][docker] (latest stable version)
+ - [Docker compose][docker-compose]
+ - [npm][npm] >= 6.0.0
```
Comment (ArchangelX360) [inline docs/dev-stack.md]: this is the Node requirement
npm is only ^4.4.4

File: docs/dev-stack.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+repository named as `bazel/powerspikegg:xxx` where `xxx` is the service run.
+It will also generate a `docker-compose.yml` able to start the whole
+development stack.
+
+    devutils/generate-docker-compose
+
+# Bringing the development stack alive
+
+To use the development stack, you now just have to move to the root workspace
+and bring the docker composed stack.
```
Comment (ArchangelX360) [inline docs/dev-stack.md]: bring the docker composed stack **up**.

**Review by ArchangelX360 — CHANGES_REQUESTED** [review]

File: docs/dev-stack.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+# Building the development stack
+
+We use bazel to generate docker images required to run the development stack.
+The following script will create a set of docker images in your local
+repository named as `bazel/powerspikegg:xxx` where `xxx` is the service run.
+It will also generate a `docker-compose.yml` able to start the whole
+development stack.
+
+    devutils/generate-docker-compose
```
Comment (ArchangelX360) [inline docs/dev-stack.md:42]: ./devutils/generate-docker-compose

**Comment by FunkySayu** [comment]
Should be good

**Review by ArchangelX360 — APPROVED** [review]

**Review by bouillonpi — APPROVED** [review]

**Review by cassand — APPROVED** [review]

**Pull request merged**